src/ffi/setobject.rs

```diff
@@ -46,12 +29,78 @@ pub struct PySetObject {
     pub weakreflist: *mut PyObject,
 }
 
+// skipped PySet_GET_SIZE
+
+// skipped _PySet_Dummy
+
+extern "C" {
+    #[cfg(not(Py_LIMITED_API))]
+    #[cfg_attr(PyPy, link_name = "_PyPySet_NextEntry")]
+    pub fn _PySet_NextEntry(
+        set: *mut PyObject,
+        pos: *mut Py_ssize_t,
+        key: *mut *mut PyObject,
+        hash: *mut super::Py_hash_t,
+    ) -> c_int;
+
+    // skipped non-limited _PySet_Update
+}
+
+#[cfg_attr(windows, link(name = "pythonXY"))]
+extern "C" {
+    #[cfg_attr(PyPy, link_name = "PyPySet_Type")]
+    pub static mut PySet_Type: PyTypeObject;
+    #[cfg_attr(PyPy, link_name = "PyPyFrozenSet_Type")]
+    pub static mut PyFrozenSet_Type: PyTypeObject;
+    pub static mut PySetIter_Type: PyTypeObject;
+}
+
+extern "C" {
+    #[cfg_attr(PyPy, link_name = "PyPySet_New")]
+    pub fn PySet_New(arg1: *mut PyObject) -> *mut PyObject;
+    #[cfg_attr(PyPy, link_name = "PyPyFrozenSet_New")]
+    pub fn PyFrozenSet_New(arg1: *mut PyObject) -> *mut PyObject;
+
+    #[cfg_attr(PyPy, link_name = "PyPySet_Add")]
+    pub fn PySet_Add(set: *mut PyObject, key: *mut PyObject) -> c_int;
+    #[cfg_attr(PyPy, link_name = "PyPySet_Clear")]
+    pub fn PySet_Clear(set: *mut PyObject) -> c_int;
+    #[cfg_attr(PyPy, link_name = "PyPySet_Contains")]
+    pub fn PySet_Contains(anyset: *mut PyObject, key: *mut PyObject) -> c_int;
+    #[cfg_attr(PyPy, link_name = "PyPySet_Discard")]
+    pub fn PySet_Discard(set: *mut PyObject, key: *mut PyObject) -> c_int;
+    #[cfg_attr(PyPy, link_name = "PyPySet_Pop")]
+    pub fn PySet_Pop(set: *mut PyObject) -> *mut PyObject;
+    #[cfg_attr(PyPy, link_name = "PyPySet_Size")]
+    pub fn PySet_Size(anyset: *mut PyObject) -> Py_ssize_t;
+}
+
+#[cfg_attr(windows, link(name = "pythonXY"))]
+extern "C" {
```

Comment:
Usually the `windows` link name attribute is only needed for `static mut` symbols, you might be able to merge this `extern` block with the one above:

```suggestion
```

Reply:
> Usually the `windows` link name attribute is only needed for `static mut` symbols
Could you please elaborate why?
In my understanding, `link` statement will be emitted in build script. Why this is necessary to add the attribute only to `static mut` symbols but not function?
Thanks in advance. I couldn't understand this after researching for several hours.

>, you might be able to merge this `extern` block with the one above:
Will do.

Reply:
Yeah, unfortunately I haven't found this documented anywhere. I suspect it's not intended, however in practice I've found that we experience windows linker errors for `static mut` symbols if they aren't covered explicitly by a `link_name` attribute. I should really report this upstream...